Blog for 0.7 release (#49)
kserve/website: docs/blog/articles/2021-10-11-KServe-0.7-release.md

---
title: "The KServe 0.7 release ensures a smooth transition from KFserving to KServe"
date: 2020-10-11
description: "KServe 0.7 Release Blog Post"
type: "blog"
---

### Authors
 **Dan Sun**, **Animesh Singh**, **Vedant Padwal** on behalf of the KServe Working Group.

### **What's Changed?**

- #### [Migration job for existing KFServing installation](/admin/migration)
- `InferenceService` API group is changed from `serving.kubeflow.org` to `serving.kserve.io` [#1826](https://github.com/kserve/kserve/issues/1826).

> - Python SDK name is changed from `kfserving` to `kserve` [Python SDK for KServe Server and Client](https://pypi.org/project/kserve).

Reviewer: ```suggestion
- Python SDK name is changed from [kfserving](https://pypi.org/project/kfserving) to [kserve](https://pypi.org/project/kserve).
```

- KServe Installation manifests [#1824](https://github.com/kserve/kserve/issues/1824).

> - Model-web-app is separated out of the kserve repository to [models-web-app](https://github.com/kserve/models-web-app).

Reviewer: ```suggestion
- Models-web-app is separated out of the kserve repository to [models-web-app](https://github.com/kserve/models-web-app).
```

> - Docs and examples are moved to separate repository https://github.com/kserve/website.

Reviewer: ```suggestion
- Docs and examples are moved to separate repository [website](https://github.com/kserve/website).
```

- KServe images are migrated to kserve docker hub account.
- v1alpha2 API group is deprecated [#1850](https://github.com/kserve/kserve/issues/1850).

### **🌈 What's New?**

- #### ModelMesh project is joining KServe under repository [modelmesh-serving](https://github.com/kserve/modelmesh-serving)!
    ModelMesh is designed for high-scale, high-density and frequently-changing model use cases.
    ModelMesh intelligently loads and unloads AI models to and from memory to strike an intelligent 
    trade-off between responsiveness to users and computational footprint.
    <br>
        To learn more about ModelMesh features and components, 

>         check out the ModelMesh announcement blog. [ModelMesh release blog](https://developer.ibm.com/blogs/kserve-and-watson-modelmesh-extreme-scale-model-inferencing-for-trusted-ai)

Reviewer: ```suggestion
        check out the [ModelMesh announcement blog](https://developer.ibm.com/blogs/kserve-and-watson-modelmesh-extreme-scale-model-inferencing-for-trusted-ai)
```

    <br>
     [Join talk at #KubeCon NA tomorrow to get a deeper dive into ModelMesh and KServe.](https://www.linkedin.com/feed/update/urn:li:activity:6854064203360280576/)
<br>  
- #### (Alpha feature) Raw kubernetes deployment support, Istio/Knative dependency is now optional [Installation guide](/admin/kubernetes_deployment).

>     The raw kubernetes deployment is for lightweight dependency and unlocks the Knative limitations.

Reviewer: ```suggestion
    The raw Kubernetes deployment is for lightweight dependency and it unlocks the Knative limitations.
```

<br>

> - KServe now has its own website temporarily hosted on [website](https://kserve.github.io/website)

Reviewer: ```suggestion
- KServe now has its own documentation website temporarily hosted on [website](https://kserve.github.io/website)
```

---
title: "The KServe 0.7 release ensures a smooth transition from KFserving to KServe"
date: 2020-10-11
description: "KServe 0.7 Release Blog Post"
type: "blog"
---

### Authors

>  **Dan Sun**,**Animesh Singh**,**Vedant Padwal** on behalf of the KServe Community

Reviewer: ```suggestion
 **Dan Sun**, **Animesh Singh**, **Vedant Padwal** on behalf of the KServe Working Group.
```

### **What's Changed?**

- #### [Migration job for existing KFServing installation](https://kserve.github.io/website/admin/migration)
    This helps to migrate existing inference services running in the cluster from KFServing to KServe without downtime. 

>     The script installs `KServe` from 0.7 release and deletes your `KFServing` installation after migrating the inference services from `serving.kubeflow.org` to `serving.kserve.io`.

Reviewer: Do not need to repeat the docs here as you already provided the link, I think the migration instruction still needs to be tweaked.

<br>
- `InferenceService` API group is changed from `serving.kubeflow.org` to `serving.kserve.io` [#1826](https://github.com/kserve/kserve/issues/1826).

> - Python SDK name is changed from `kfserving` to `kserve` [#1827](https://github.com/kserve/kserve/issues/1827).

Reviewer: Can you help provide the link to the pypi package https://pypi.org/project/kserve ?

- KServe Installation manifests [#1824](https://github.com/kserve/kserve/issues/1824).

> - Model-web-app is separated out of the kserve repository to [#1820](https://github.com/kserve/kserve/issues/1820).

Reviewer: to `https://github.com/kserve/models-web-app`

- Docs and examples are moved to separate repository https://github.com/kserve/website.
- KServe images are migrated to kserve docker hub account.
- v1alpha2 API group is deprecated [#1850](https://github.com/kserve/kserve/issues/1850).

### **🌈 What's New?**

- #### ModelMesh project is joining KServe under repository [modelmesh-serving](https://github.com/kserve/modelmesh-serving)!
    ModelMesh is designed for high-scale, high-density and frequently-changing model use cases.
    ModelMesh intelligently loads and unloads AI models to and from memory to strike an intelligent 
    trade-off between responsiveness to users and computational footprint.
        To learn more about ModelMesh features and components, check out the ModelMesh announcement blog. [ModelMesh release blog](https://developer.ibm.com/blogs/kserve-and-watson-modelmesh-extreme-scale-model-inferencing-for-trusted-ai/)
<br>  

> - #### (Alpha feature) Raw kubernetes deployment support,Istio/Knative dependency is now optional [#1508](https://github.com/kserve/kserve/issues/1508),[#1830](https://github.com/kserve/kserve/issues/1830).

Reviewer: ```suggestion
- #### (Alpha feature) Raw kubernetes deployment support, Istio/Knative dependency is now optional [#1508](https://github.com/kserve/kserve/issues/1508),[#1830](https://github.com/kserve/kserve/issues/1830).
```

Reviewer: Can point to the installation guide https://kserve.github.io/website/admin/kubernetes_deployment

    KFServing totally depends on Knative to bring serving features like scale to zero, Knative Pod Autoscaler(KPA)
    and canary rollouts to ML/DL deployments, and the serving runtime contract requires HTTP as the inboundtransport.
    There are use cases where Knative would be a great solution but where HTTP is not used. Especially, due to safety 
    concerns, some cases need alternative transports like UDP or TCP or even custom protocols. So, we need an effective 

>     scenario to solve the problem, make Knative Serving and native serving should be both available for the Kubernetes cluster.

Reviewer: This is not quite accurate which needs to be rephrased. The raw kubernetes deployment is for lightweight dependency and unlocks the Knative limitations.

<br>
- KServe now has its own website temporarily hosted on [website](https://kserve.github.io/website)
- Support v1 crd and webhook configuration for Kubernetes 1.22 [#1837](https://github.com/kserve/kserve/issues/1837).
- Triton model serving runtime now defaults to 21.09 version [#1840](https://github.com/kserve/kserve/issues/1840).

### **🐞 What's Fixed?**

> - Bug fix for Azure blob

Reviewer: link to the issue?
